SPECS/application-gateway-kubernetes-ingress/application-gateway-kubernetes-ingress.spec

```
@@ -24,20 +24,21 @@ Source0:        %{name}-%{version}.tar.gz
 #           -cf %%{name}-%%{version}-vendor.tar.gz vendor
 #
 Source1:        %{name}-%{version}-vendor.tar.gz
-
+# If upstream ever upgrades client_goland to 1.11.1, we can get rid of this patch.
+Patch0:         CVE-2022-21698.patch
 BuildRequires:  golang >= 1.13
 
 %description
 This is an ingress controller that can be run on Azure Kubernetes Service (AKS) to allow an Azure Application Gateway 
 to act as the ingress for an AKS cluster. 
 
 %prep
-%autosetup
-
-%build
+%setup
 rm -rf vendor
 tar -xf %{SOURCE1} --no-same-owner
+%patch -P 0 -p 1 -d vendor/github.com/prometheus/client_golang
```

Would like opinions about doing this vs. having that extra path info in the patch itself.

---

I personally like it better (at least after a short 2 minute consideration), because it makes it easier to see which patches are for the upstream source and which patch the vendor tarballs.

One thing, though, to align with how we use `%patch` in other specs:
```suggestion
%patch0 -p1 -d vendor/github.com/prometheus/client_golang
```

---

I considered doing that, but... https://rpm-software-management.github.io/rpm/manual/spec.html lists ways to apply a patch (in the examples, `Patch1`), and says this...
```
%patch 1 (since rpm >= 4.18)
%patch -P1 (all rpm versions)
%patch1 (deprecated, do not use)
```

Should I use the deprecated version for the sake of consistency?

---

I was not aware of that, thank you for letting me know. In that case your way is better.

---

Per offline discussion, changed to `%patch 0 -p1 ...`